File: src/main/java/io/bindernews/bnsts/IVariable.java
package io.bindernews.bnsts;

import com.megacrit.cardcrawl.cards.AbstractCard;
import org.jetbrains.annotations.NotNull;

public interface IVariable {

    int value(@NotNull AbstractCard card);
    int baseValue(@NotNull AbstractCard card);
    boolean upgraded(@NotNull AbstractCard card);
    boolean isModified(@NotNull AbstractCard card);

    void setValue(@NotNull AbstractCard card, int amount);
    void setBaseValue(@NotNull AbstractCard card, int amount);
    void upgrade(@NotNull AbstractCard card, int amount);
}

File: src/main/kotlin/net/bindernews/grackle/helper/IVariable.kt
package net.bindernews.grackle.helper

import com.megacrit.cardcrawl.cards.AbstractCard

interface IVariable {
    fun value(card: AbstractCard): Int
    fun baseValue(card: AbstractCard): Int
    fun upgraded(card: AbstractCard): Boolean
    fun isModified(card: AbstractCard): Boolean
    fun setValue(card: AbstractCard, amount: Int)
    fun setBaseValue(card: AbstractCard, amount: Int)
    fun upgrade(card: AbstractCard, amount: Int)
}